subsys/bluetooth/host/hci_core.h

```diff
@@ -305,6 +313,30 @@ extern const struct bt_conn_auth_cb *bt_auth;
 enum bt_security_err bt_security_err_get(uint8_t hci_err);
 #endif /* CONFIG_BT_SMP || CONFIG_BT_BREDR */
 
+/* Data type to store state related with commnad to be updated
+ * when command completes successfully.
+ */
+struct cmd_state_set {
+	/* Target memory to be updated */
+	atomic_t *target;
+	/* Bit number to be updated in target memory */
+	int bit;
+	/* Value to determine if enable or disable bit */
+	bool val;
+};
+
+/* Initialize commnad state instance */
+static inline void cmd_state_set_init(struct cmd_state_set *state,
+				      atomic_t *target, int bit, bool val)
+{
+	state->target = target;
+	state->bit = bit;
+	state->val = val;
+}
+
+/* Set commnad state for given network buffer */
+void cmd_data_state_set(struct net_buf *buf, struct cmd_state_set *state);
```

Comment:
Sorry, I should have commented on this as well. Even though it's an internal header file, `hci_core.h` has had a strict `bt_*` namespace so far, and it seems you're the first one to try to introduce content to it that violates that. It's not really a big deal since these are internal files, but it will cause some confusion when starting to look for the definitions, since when I don't see a `bt_*` prefix I'd assume that they are located in the same c-file.

Reply:
Changed as suggested. I thought that only public API has bt_ prefix. What about hci_ (e.g. hci_evt_pin_code_req)? Is it an exception for HCI event handlers?

Reply:
If you look at the header files under `subsys/bluetooth/host/*.h` (which are all internal, public files are found under `include/bluetooth`) you'll see that pretty much everything follows the `bt_*` namespace

Reply:
We can argue about whether it makes sense, but the higher priority IMO is to keep things consistent

Reply:
I was not going to argue ;-) I was just lecturing about conventions in Host implementation.